instrumentation/liberty/liberty/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/liberty/LibertyInstrumentationModule.java

```diff
@@ -20,6 +20,20 @@
 import net.bytebuddy.description.type.TypeDescription;
 import net.bytebuddy.matcher.ElementMatcher;
 
+/**
+ * Instrumenting request handling in liberty.
+ *
+ * <ul>
+ *   <li>On entry to WebApp.handleRequest remember request.
+ *   <li>On call to WebApp.isForbidden (called from WebApp.handleRequest) start span based on
+ *       remembered request. We don't start span immediately at the start or handleRequest because
+ *       HttpServletRequest isn't usable yet.
+ *   <li>On exit from WebAppDispatcherContext.setPathElements (called from WebApp.handleRequest)
+ *       update span name. We don't do it before because before this method is called we can't use
+ *       HttpServletRequest.getServletPath.
+ *   <li>On exit from WebApp.handleRequest close the span.
+ * </ul>
+ */
```

Comment:
Thanks! This is really helpful. Can you add links to corresponding advice classes as well?

Reply:
Done.